cloudinit/userdata_cloudinit.erb

```diff
@@ -41,6 +49,28 @@ users:
    lock-passwd: false
    passwd: <%= @host.root_pass %>
 
+ runcmd:
+ <% if rhel_compatible -%>
+ - <%= snippet('epel') -%>
+ <% end -%>
+ <% if puppet_enabled -%>
+ - <%= snippet('puppetlabs_repo') -%>
+ <% end -%>
+ - |
+ <%= indent(2) { snippet('remote_execution_ssh_keys') } %>
+ <% if chef_enabled %>
+ - |
+ <%= indent(2) { snippet('chef_client') } %>
+ <% end -%>
+ <% if puppet_enabled %>
```

Comment:
Just combine the two puppet_enabled sections, configure the repo and puppet_setup in the same bit.